### PR TITLE
Fix spacing/formatting/issues in altos-core and libs/volatile

### PR DESCRIPTION
**Pull request by RJ-Russell**
@pdouglas94 
@dnseitz

### CODE REVIEW

**Comment by RJ-Russell** [comment]
I'm good with this. :+1:  I'll hold off approving it until @pdouglas94  has taken a look as well.

File: altos-rust/altos-core/libs/heap/free_list_allocator/src/free_list.rs
Comment (pdouglas94) [inline altos-rust/altos-core/libs/heap/free_list_allocator/src/free_list.rs]: Why was this changed? I feel like indenting this makes it more clear that it's not a separate statement, but part of the assignment. The style guide doesn't seem to specify how this should be done.

File: altos-rust/altos-core/libs/heap/free_list_allocator/src/free_list.rs
Comment (pdouglas94) [inline altos-rust/altos-core/libs/heap/free_list_allocator/src/free_list.rs]: Same as above. I'm not sure what the right way to indent this would be, but it looks kind of funny to me having it on the same level of indentation as `fn find_block`.

Comment (dnseitz) [reply altos-rust/altos-core/libs/heap/free_list_allocator/src/free_list.rs]: I think we just missed this when we were fixing the indentation in all the files. The auto-indent seemed to like to put the two lines at the same indentation level.

File: altos-rust/altos-core/libs/heap/free_list_allocator/src/lib.rs
Comment (pdouglas94) [inline altos-rust/altos-core/libs/heap/free_list_allocator/src/lib.rs:68]: These functions are part of the required API for custom allocators (https://doc.rust-lang.org/book/custom-allocators.html). It would probably be better to make the argument names match the API exactly. For deallocate, it looks like it's actually `__rust_deallocate(ptr: *mut u8, _old_size: usize, _align: usize)`. We should check the other API functions as well.

File: altos-rust/altos-core/libs/heap/free_list_allocator/src/test.rs
Comment (pdouglas94) [inline altos-rust/altos-core/libs/heap/free_list_allocator/src/test.rs]: Having the curly bracket on the next line isn't the way we're doing it in other places, like `free_list.rs`, so this seems like inconsistent style.

File: altos-rust/altos-core/src/arch/cm0.rs
Comment (pdouglas94) [inline altos-rust/altos-core/src/arch/cm0.rs]: Shouldn't the stuff inside `asm!` be indented another level?

File: altos-rust/altos-core/src/arch/cm0.rs
Comment (pdouglas94) [inline altos-rust/altos-core/src/arch/cm0.rs]: Same as above.

File: altos-rust/altos-core/libs/heap/free_list_allocator/src/lib.rs
Comment (dnseitz) [reply altos-rust/altos-core/libs/heap/free_list_allocator/src/lib.rs:68]: The underscore in front of a variable name marks it as unused, so we don't get compiler warnings for unused variables. The doc examples show them as wrappers around a C style malloc library that only takes the pointer to the memory to be freed, and doesn't use the `size` or `align` parameters. In our case, however, we are using those parameters so they shouldn't have an underscore in front of them.

File: altos-rust/altos-core/libs/heap/free_list_allocator/src/test.rs
Comment (dnseitz) [reply altos-rust/altos-core/libs/heap/free_list_allocator/src/test.rs]: Hm, yeah, we want to keep it the same. Personally I like the bracket on it's own line if the function signature extends across multiple lines so it's clear where the signature ends and the function body begins. Any objections to adopting that as style?

File: altos-rust/altos-core/src/task/control.rs
Comment (pdouglas94) [inline altos-rust/altos-core/src/task/control.rs]: These appear to be helper functions, not tests themselves. Changing the name has caused compiler errors because these functions are being called by a bunch of the other tests. It might make sense to move this out to test.rs, or if it's left here, maybe add an underscore to the name like `_get_task`. Either way, all the other test functions that are calling this should be updated.

File: altos-rust/altos-core/src/task/control.rs
Comment (pdouglas94) [inline altos-rust/altos-core/src/task/control.rs]: My comment about `get_task` above applies to this one as well.

File: altos-rust/altos-core/libs/heap/free_list_allocator/src/test.rs
Comment (RJ-Russell) [reply altos-rust/altos-core/libs/heap/free_list_allocator/src/test.rs]: So move the curly brace? or leave it? @dnseitz  and I actually had a discussion... it really doesn't matter to me either way.. just need to know what the final decision is.

Comment (pdouglas94) [reply altos-rust/altos-core/libs/heap/free_list_allocator/src/test.rs]: It is probably more clear this way. On the other hand, it seems sort of inconsistent since most of the time we have brackets, we're doing:
```
fn/if/something {
}
```
and this would be the only case where we're doing it differently. I think allowing it would also be another exception we're making to the specified style guide. I'm okay with allowing it though.

**Review by pdouglas94 — APPROVED** [review]

**Pull request merged**
